pages/projects/[id].tsx

```diff
@@ -45,32 +52,29 @@ const Project: NextPage = (): JSX.Element => {
   `;
 
   const router = useRouter();
-  const { id }: UrlParams = router.query;
+  const { id, story: selectedStoryId }: UrlParams = router.query;
   const dispatch = useDispatch();
   const [filters, setFilters] = useState<Filters>({});
   const [unestimatedStory, setUnestimatedStory] = useState<Story>();
   const apiKey = useSelector(getApiKey);
   const stories = useSelector(
     (state: State): Record<string, Story[]> => filterStories(state, id, filters)
   );
-
-  const getFilterArray = (filters: Owner[] | Label[] = [], filter: Owner | Label) => {
-    if (filters.find(f => f.name === filter.name)) {
-      return filters;
-    }
-    return [...filters, filter];
-  };
-
-  const addFilter = (name: string, filter: Owner | Label | Iteration): void => {
-    if (name === 'iterations') {
-      //todo: we should change this check to a type check and uses classes so we don't have to do this.
-      // @ts-ignore: we know iteration can only be Iteration type
-      setFilters({ ...filters, iteration: filter });
-      return;
-    }
-    // @ts-ignore: we know filter cannot be type Iteration
-    setFilters({ ...filters, [name]: getFilterArray(filters[name], filter) });
-  };
+  const selectedStory = useSelector((state: State): Story => getStory(state, id, selectedStoryId));
+
+  const addFilter = useCallback(
+    (name: string, filter: Owner | Label | Iteration): void => {
+      if (name === 'iterations') {
+        //todo: we should change this check to a type check and uses classes so we don't have to do this.
+        // @ts-ignore: we know iteration can only be Iteration type
+        setFilters({ ...filters, iteration: filter });
```

Comment:
Think you can use type predicates or `[instanceof](https://www.typescriptlang.org/docs/handbook/advanced-types.html#instanceof-type-guards)` operation to type guard
```suggestion
        setFilters({ ...filters, iteration: (filter as Iteration) });
```